app/react/src/client/preview/render.tsx

```diff
@@ -26,10 +31,56 @@ export const render: ArgsStoryFn<ReactFramework> = (args, context) => {
   return <Component {...args} />;
 };
 
-const renderElement = async (node: ReactElement, el: Element) =>
-  new Promise((resolve) => {
-    ReactDOM.render(node, el, () => resolve(null));
+const renderElement = async (node: ReactElement, el: Element) => {
+  // Create Root Element conditionally for new React 18 Root Api
+  const root = await getReactRoot(el);
+
+  return new Promise((resolve) => {
+    if (root) {
+      root.render(node);
+      setTimeout(() => {
+        resolve(null);
+      }, 0);
+    } else {
+      ReactDOM.render(node, el, () => resolve(null));
+    }
   });
+};
+
+const canUseNewReactRootApi = reactDomVersion.startsWith('18');
```

Comment:
```suggestion
const canUseNewReactRootApi = reactDomVersion.startsWith('18') || reactDomVersion.startsWith('0.0.0');
```

Which adds support for `react@exprimental`.